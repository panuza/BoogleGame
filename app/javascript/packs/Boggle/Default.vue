<template>
  <div id="boggle-app" class="text-center">
    <p class="pt-5">Welcome to boggle game!!!</p>
    <div class="row">
      <div class="col-md-2 offset-md-5">
       <!--  <form class="text-center pt-3" accept-charset="UTF-8" @submit="showBoard"> -->
        <div class="form-group mx-sm-3 mt-5 mb-2">
          <label for="name" class="sr-only text-center">Password</label>
          <input type="text" class="form-control" id="name" placeholder="Enter your name" v-model="user_name" required="true">
        </div>
        <button class="btn btn-primary mb-2 mt-3" @click="showBoard">Start Game</button>
      <!-- </form> -->
      </div>
    </div>
  </div>
</template>

<script>
import axios from 'axios'

export default {
  data() {
    return {
      user_name: '',

    }
  },

  mounted(){
  },

  methods: {
    showBoard(){
      axios({
              method: "post",
              url: '/api/user_details.json',
              params: { name: this.user_name},
              withCredentials: false
            })
      .then(res => {
        window.location.href = "/board"
      })
      .catch(error => {});

    }
  }

}
</script>

<style scoped>
p {
  font-size: 2em;
  text-align: center;
  color: red;
}

.text-center{
  text-align: center;
}

.bg-color{
  background-color: #EBFC7E;
}

.vertical-center{
  vertical-align: ""
}
</style>
