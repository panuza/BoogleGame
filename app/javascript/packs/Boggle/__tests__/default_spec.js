import { mount } from '@vue/test-utils'
import Default from '../../Boggle/Default.vue'

describe('Default', () => {
  test('renders correctly', () => {
    const wrapper = mount(Default)
    expect(wrapper.element).toMatchSnapshot()
  })
})
