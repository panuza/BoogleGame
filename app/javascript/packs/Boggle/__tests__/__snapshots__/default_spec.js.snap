// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default renders correctly 1`] = `
<div
  class="text-center"
  id="boggle-app"
>
  <img
    alt="avatar"
    src="https://process.filestackapi.com/cache=expiry:max/resize=width:700/GliZS1I9QHKlu0FPNocW"
  />
   
  <div
    id="boggle-start"
  >
    <div
      class="text-left"
    >
      <div
        class="row"
      >
        <div
          class="col-md-4 offset-md-4"
        >
          <h1
            class="pt-5 text-left"
          >
            Welcome to boggle game!!!
          </h1>
           
          <span
            style="font-size: 14px;"
          >
            
          Boggle game is played using a grid of letters, in which the players attempt to find words in sequences of adjacent letters,
          which is a word game invented by Allan Turoff and originally distributed by Parker Brothers. 
          
            <br />
            
          Basically, The player searches for words that can be constructed from the letters of sequentially adjacent cubes, where "adjacent" cubes are those horizontally, vertically and diagonally neighboring. Words must be at least three letters long, may include singular and plural (or other derived forms) separately. The application records all the words that is submitted by the player. After three minutes have elapsed, the application stops the player from the game play and the game displays the total score.
        
          </span>
        </div>
      </div>
    </div>
     
    <div
      class="row"
    >
      <div
        class="col-md-4 offset-md-4"
      >
        <div
          class="form-group mx-sm-3 mt-5 mb-2"
        >
          <label
            class="sr-only text-center"
            for="name"
          >
            Password
          </label>
           
          <input
            class="form-control"
            id="name"
            placeholder="Enter your name"
            required="required"
            type="text"
          />
        </div>
         
        <button
          class="btn btn-success w-50 mt-5"
        >
          Start Game
        </button>
      </div>
    </div>
  </div>
   
  <!---->
</div>
`;
